src/terminal/adapter/InteractDispatch.cpp

```diff
@@ -59,18 +59,8 @@ void InteractDispatch::WriteCtrlKey(const INPUT_RECORD& event)
 // - string : a string to write to the console.
 void InteractDispatch::WriteString(const std::wstring_view string)
 {
-    if (!string.empty())
-    {
-        const auto codepage = _api.GetConsoleOutputCP();
-        InputEventQueue keyEvents;
-
-        for (const auto& wch : string)
-        {
-            CharToKeyEvents(wch, codepage, keyEvents);
-        }
-
-        WriteInput(keyEvents);
-    }
+    const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    gci.GetActiveInputBuffer()->WriteString(string);
 }

```

Comment:
You're going to hate me for bringing this up, but I think we might actually need both versions of this function, with `ActionPrintString` calling the old version (i.e. generating key events), and `ActionPassThroughString` calling your new version that writes directly to the input buffer.

The reason being that we still need to support third-party terminals that don't necessarily implement win32-input-mode, and you can see how that affects things by commenting out our implementation of the win32-input-mode in `AdaptDispatch`.

With this PR applied, regular character keys no longer generate up and down key events with their actual virtual key code - you just get a single null virtual key code with the appropriate char value. That may be good enough in many cases, but I suspect there will be some apps that require full key events.

So for that to work correctly, I think `ActionPrintString` needs to be left as it was using the old implementation of `WriteString`. And to fix the VT responses, we just need to make `ActionPassThroughString` call your new version.

Reply:
Oh... of course. But this function also gets called during a bracketed paste, right? That's unfortunate but I think that's a minor issue for later. As I mentioned previously, I was planning to rework the `InputBuffer` until the next release, and that would also entail moving the conversion from plain text into `INPUT_RECORD`s until when the data gets read and not when it's being written. That will fix various subtle issues like this one, I'm sure.

Reply:
I'm not sure if I've misunderstood your intention here, but I thought we were going to keep your new version of `WriteString` for the `ActionPassThroughString`, because that's what was fixing the VT responses. I've checkout your last revert now, and it seems that yazi doesn't work anymore, and a `DA1` query from the cmd prompt no longer shows the initial escape. Is the plan to address that later?

Reply:
Ok, I shouldn't have done this on a holiday. 😓 I'll run some tests now and get this right.

Reply:
Fixed & tested with yazi. Only `ActionPassThroughString` calls the new code now, while the old `ActionPrintString` continues to call `InputBuffer::Write`.